LayoutTests/http/tests/site-isolation/inspector/target/target.html

```diff
@@ -0,0 +1,35 @@
+<script src="../../../inspector/resources/inspector-test.js"></script>
+<script>
+    function test() {
+        let suite = InspectorTest.createAsyncSuite("site-isolation.inspector.target");
+        suite.addTestCase({
+            name: "site-isolation.inspector.target.target-lifecycle-events",
+            description: "Target lifecycle events",
+            test(resolve) {
+                let eventsSeen = 0;
+                const eventsExpected = 4;
+                WI.targetManager.addEventListener(WI.TargetManager.Event.TargetAdded, function (event) {
+                    InspectorTest.log(`targetAdded: type ${event.data.target.type}`);
+                    ++eventsSeen;
+                });
+                WI.targetManager.addEventListener(WI.TargetManager.Event.TargetRemoved, function (event) {
+                    InspectorTest.log(`targetRemoved: type ${event.data.target.type}`);
+                    ++eventsSeen;
+                    if (eventsSeen == eventsExpected)
+                        resolve();
+                });
+                InspectorTest.evaluateInPage(`
+                    let iframe = document.createElement("iframe");
+                    iframe.src = "http://127.0.0.1:8000/site-isolation/inspector/target/resources/middle-iframe.html";
+                    document.body.appendChild(iframe);
+                    setTimeout(function () {
+                        document.body.removeChild(iframe);
+                    }, 100);
```

Comment:
Recommend against using hardcoded timeouts. I'd recommend having one `eval` to set up the page contents, then a second `eval` to actually trigger the behavior under test.